src/redisearch_api.c

```diff
@@ -179,6 +179,18 @@ void RediSearch_DocumentAddFieldNumber(Document* d, const char* fieldname, doubl
   Document_AddFieldC(d, fieldname, buf, len, as);
 }
 
+void RediSearch_DocumentAddFieldGeo(Document* d, const char* fieldname, 
+                                    double lat, double lon, unsigned as) {
+  if (lat > GEO_LAT_MAX || lat < GEO_LAT_MIN || lon > GEO_LONG_MAX || lon < GEO_LONG_MIN) {
+    // out of range
+    return;
+  }                                      
+  // The format for a geospacial point is "lat,lon"
+  char buf[24];
+  size_t len = sprintf(buf, "%.6lf,%.6lf", lat, lon);
```

Comment:
I thought we wanted to avoid string representation, as this string is probably broken down again to the individual components later on

Reply:
@swilly22  There is no simple way around it since the document fields got through the preprocessor which expects a string.